include/h2o/http2_internal.h

```diff
@@ -170,6 +170,14 @@ struct st_h2o_http2_conn_t {
         h2o_linklist_t blocked_streams;
     } early_data;
     h2o_iovec_t *http2_origin_frame;
+#define HTTP2_OLD_PRIORITIES 10
+    struct {
+        struct {
+            h2o_http2_scheduler_openref_t *sched_node;
+            uint32_t stream_id;
```

Comment:
I notice that in `get_parent_stream_id` function of lib/http2/connection.c, we might refer to `h2o_http2_stream_t::stream_id` that is relative to the address of `sched_node`.

Considering that, it might make more sense to simply keep a pointer to `h2o_http2_stream_t` that has all the values disposed of with the exception being `stream_id` and `_refs.scheduler`, rather than retaining a pointer to `sched_node` as well as separately keeping `stream_id`.

Note however that we also agree that it's a bit of waste to retain `h2o_http2_stream_t` even after the stream is closed. FWIW, I think we could use something like 94536be to move just the scheduling node to a different location.

The way of adjusting this structure depends on if and how we might fix the issue.